File: backend/src/main/java/ru/sber/services/OrderService.java
package ru.sber.services;

import org.springframework.http.ResponseEntity;
import ru.sber.models.Order;

import java.util.List;

/**
 * Получает информацию о заказе
 */
public interface OrderService {
    /**
     * Обновляет статус заказа
     *
     * @param id    id заказа
     * @param order заказ
     * @return Результат
     */
    ResponseEntity<?> updateOrderStatusById(Long id, Order order);

    /**
     * Отменяет заказ
     *
     * @param id      id заказа
     * @param massage сообщение
     * @return Результат
     */
    ResponseEntity<?> cancelOrderById(Long id, Object massage);

    /**
     * Отменяет заказы
     *
     * @param listId listId заказов
     * @return Результат
     */
    ResponseEntity<?> cancelOrderByListId(String listId, Object massage);

    /**
     * Получает информацию о заказах
     *
     * @return Лист заказов
     */
    List<?> getListOrders();

    /**
     * Получает информацию о заказах после уведомления
     *
     * @return Лист заказов
     */
    List<?> getListOrdersByNotify();
}

File: frontend/src/services/orderService.ts
import axios from "axios";
import {Dispatch} from "redux";
import authHeader from "./auth-header";
import {setAllOrders, updateOrders} from "../slices/orderSlice";
import {AccountUser} from "../types/accountType";


const API_URL_ORDER = "orders"

async function updateOrderStatusById(id: number, status: string, dispatch: Dispatch) {
    const headers = authHeader();
    try {
        console.log(id)
        const response = await axios.put(API_URL_ORDER + `/${id}`,
            {status: status},
            {headers});

        return response.data;

    } catch (error) {
        console.error("Ошибка обновления заказа:", error);
        throw error;
    }
}

async function cancelOrderById(id: number, message: string, dispatch: Dispatch): Promise<AccountUser> {
    const headers = authHeader();

    try {
        const response = await axios.put(API_URL_ORDER + `/${id}/cancel`,
            {message: message},
            {headers});

        return response.data;

    } catch (error) {
        console.error("Ошибка отмены заказа:", error);
        throw error;
    }
}

async function cancelOrderByListId(id: string, message: string, dispatch: Dispatch): Promise<AccountUser> {
    const headers = authHeader();

    try {
        const response = await axios.put(API_URL_ORDER + `/cancel?listId=${id}`,
            {message: message},
            {headers});

        return response.data;

    } catch (error) {
        console.error("Ошибка отмены заказов:", error);
        throw error;
    }
}

async function getListOrders(dispatch: Dispatch) {
    const headers = authHeader();

    try {
        const response = await axios.get(API_URL_ORDER, {headers});

        dispatch(setAllOrders(response.data));
        return response.data;
    } catch (error) {
        console.error("Ошибка получения заказов:", error);
    }
}

async function getListOrdersByNotify(dispatch: Dispatch): Promise<number> {
    const headers = authHeader();

    try {
        const response = await axios.get(`${API_URL_ORDER}/notify`, {headers});
        dispatch(updateOrders(response.data));

        return response.data.length;
    } catch (error) {
        console.error("Ошибка получения заказов по уведомлению: ", error);
        throw error;
    }
}

const orderService = {
    updateOrderStatusById,
    cancelOrderById,
    cancelOrderByListId,
    getListOrders,
    getListOrdersByNotify,
};

export default orderService;